rpmio/rpmpgp_internal.c

```diff
@@ -1179,7 +1179,11 @@ rpmRC pgpVerifySignature(pgpDigParams key, pgpDigParams sig, DIGEST_CTX hashctx)
 	pgpDigAlg ka = key->alg;
 	if (sa && sa->verify) {
 	    if (sa->verify(ka, sa, hash, hashlen, sig->hash_algo) == 0) {
-		res = RPMRC_OK;
+		if ((key->saved & PGPDIG_UNTRUSTED) ||
+		    (sig->saved & PGPDIG_UNTRUSTED))
+		    res = RPMRC_NOTTRUSTED;
+		else
+		    res = RPMRC_OK;
```

Comment:
I agree with @pmatilai that we [shouldn't change the API](https://github.com/rpm-software-management/rpm/pull/1993#issuecomment-1097830727).

I think we can emit this type of information via [my proposed lint interface](https://github.com/rpm-software-management/rpm/issues/1974#issuecomment-1081779174).  We should, of course, consider returning richer error messages [when we transition to a new API](https://github.com/rpm-software-management/rpm/issues/1974#issuecomment-1081779174).

Reply:
If we return NOKEY, then I think the whole certificate is ignored, which is not what we want.

Reply:
I think the better approach is to just ignore the subkey if it appears to be revoked, i.e., drop this change.